### PR TITLE
Implement POST /opportunities endpoint

### PR DESCRIPTION
**Pull request by bickelj**
Creates a single opportunity when it is valid.

Issue #33 Implement POST /opportunities endpoint

### CODE REVIEW

**Comment by bickelj** [comment]
I see that it shows the full diff from main to this branch even though the PR is based on a previous branch. This is awkward. I suppose review #66 and #68 first and we can get those merged before looking here.

**Comment by bickelj** [comment]
This PR is intended only to cover 5350f92.

**Comment by bickelj** [comment]
Ah yes, and missing tests for the post as well.

**Comment by bickelj** [comment]
eddebd3 includes the needful tests.

**Comment by slifty** [comment]
Now that the other endpoints are merged  it would be good to rebase this -- I can do it but if you're back then it'd be cleaner for you to!

**Comment by slifty** [comment]
I used some of the patterns set in here in #76 but also added a few others -- for instance the test suite in the new PR might be worth checking out, as well as the validation mechanism.

**Comment by bickelj** [comment]
914ff96bca03e2ba20bd5dccc3d5ee5c4213b6d4 is the rebase without any other changes yet.

**Comment by bickelj** [comment]
I don't see the Codecov comment updated for commit e6d9397 but when looking at the coverage in https://github.com/PhilanthropyDataCommons/service/actions/runs/3130147251/jobs/5080065775 I see it should be OK now.

File: src/__tests__/opportunities.int.test.ts [hunk 1]
```diff
@@ -133,4 +137,111 @@ describe('/opportunities', () => {
       });
     });
   });
+
+  describe('GET /id', () => {
```
Comment (slifty) [inline src/__tests__/opportunities.int.test.ts:141]: This test is just to add coverage to a feature added before, right?  Let's put it in a separate commit if that's easy (not a big deal if not)

**Review by slifty — APPROVED** [review]
w00t looks great.

One tiny commit organization request which can be ignored or taken as desired.

File: src/__tests__/opportunities.int.test.ts (hunk 1 above)
Comment (bickelj) [reply src/__tests__/opportunities.int.test.ts:141]: Correct, good call. Doing it.

**Comment by bickelj** [comment]
https://github.com/PhilanthropyDataCommons/service/pull/88 has just the follow-up to issue #31 with tests. I don't recall how that fell through the cracks but oh well. Let's merge #88 first, then I will rebase here, then merge this.

**Comment by bickelj** [comment]
It turned out to be a bad rebase or bad merge where some (slightly different but largely the same) code got duplicated. The solution was to remove it here rather than add more in #88. This should be very close to ready.
